File: test-suite/src/test/java/io/micronaut/security/docs/basicauth/BasicAuthTest.java
package io.micronaut.security.docs.basicauth;

import io.micronaut.http.HttpRequest;
import org.junit.jupiter.api.Test;

import java.nio.charset.StandardCharsets;
import java.util.Base64;

import static org.junit.jupiter.api.Assertions.assertEquals;

class BasicAuthTest {

    @Test
    void basicAuthSetsTheAuthorizationHeaderWithBasicBase64UsernameAndPassword() {
        // tag::basicAuth[]
        HttpRequest<?> request = HttpRequest.GET("/home").basicAuth("sherlock", "password");
        // end::basicAuth[]
        String encoded = Base64.getEncoder().encodeToString("sherlock:password".getBytes(StandardCharsets.UTF_8));
        assertEquals("Basic " + encoded, request.getHeaders().get("Authorization"));
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/security/docs/basicauth/BasicAuthTest.kt
package io.micronaut.security.docs.basicauth

import io.micronaut.http.HttpRequest
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.util.Base64

internal class BasicAuthTest {

    @Test
    fun basicAuthSetsTheAuthorizationHeaderWithBasicBase64UsernameAndPassword() {
        // tag::basicAuth[]
        val request = HttpRequest.GET<Any>("/home").basicAuth("sherlock", "password")
        // end::basicAuth[]
        val encoded = Base64.getEncoder().encodeToString("sherlock:password".toByteArray())
        assertEquals("Basic $encoded", request.headers["Authorization"])
    }
}
